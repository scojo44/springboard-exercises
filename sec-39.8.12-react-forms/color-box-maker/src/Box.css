.Box {
  display: flex;
  justify-content: center;
  align-items: center;
  list-style: none;
}
