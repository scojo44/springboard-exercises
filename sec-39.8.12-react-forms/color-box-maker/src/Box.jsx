import React from "react";
import './Box.css';

const Box = ({id, width, height, color, removeMe}) => {
  function handleClick(e) {
    removeMe(id);
  }

  return (
    <li className="Box">
      <div style={{width, height, backgroundColor: color}}></div>
      <p><button onClick={handleClick} className="x-btn">X</button></p>
    </li>
  )
};

export default Box;
