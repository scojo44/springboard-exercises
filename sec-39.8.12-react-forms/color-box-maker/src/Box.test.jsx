import { render } from '@testing-library/react';
import Box from './Box';

describe('Box Tests', () => {
  it('renders without crashing', () => {
    render(<Box color="mediumpurple" width="100px" height="100px" removeMe={x => true} />);
  });

  it('matches snapshot', () => {
    const {asFragment} = render(<Box color="mediumpurple" width="100px" height="100px" removeMe={x => true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows the remove button', () => {
    const {getByText} = render(<Box color="mediumpurple" width="100px" height="100px" removeMe={x => true} />);
    const button = getByText('X');
    expect(button).toBeInTheDocument();
  });
});
