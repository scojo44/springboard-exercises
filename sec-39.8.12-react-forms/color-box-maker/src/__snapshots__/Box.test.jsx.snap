// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box Tests > matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Box"
  >
    <div
      style="width: 100px; height: 100px; background-color: mediumpurple;"
    />
    <p>
      <button
        class="x-btn"
      >
        X
      </button>
    </p>
  </li>
</DocumentFragment>
`;
